src/client/setProtocol.js

```diff
@@ -24,15 +25,17 @@ module.exports = function (client, options) {
         nextState: 2
       })
       client.state = states.LOGIN
+
       client.write('login_start', {
         username: client.username,
-        signature: client.profileKeys
+        playerUUID: mcData.supportFeature('signatureOnLogin') ? client.uuid : undefined,
+        signature: (mcData.supportFeature('signatureOnLogin') && client.profileKeys)
           ? {
-              timestamp: BigInt(client.profileKeys.expiresOn.getTime()), // should probably be called "expireTime"
+              timestamp: BigInt(client.profileKeys.expireTime),
               publicKey: client.profileKeys.publicDER,
-              signature: client.profileKeys.signature
+              signature: mcData.supportFeature('profileKeySignatureV2') ? client.profileKeys.signatureV2 : client.profileKeys.signature
             }
```

Comment:
missing `profileId` field in login_start

Reply:
It has playerUUID, which is the same thing. My minecraft-data is not in sync with yours

Reply:
Yeah, missed the above field